### PR TITLE
Simpler vejoe staking

### PR DESCRIPTION
**Pull request by LouisMeMyself**

### CODE REVIEW

File: contracts/VeJoeStaking.sol [hunk 1]
```diff
@@ -244,14 +145,14 @@ contract VeJoeStaking is Initializable, OwnableUpgradeable {
             userInfo.balance >= _amount,
             "VeJoeStaking: cannot withdraw greater amount of JOE than currently staked"
         );
+        update();
+        //We don't need to claim as veJOE balance will be reseted to 0
```
Comment (jummy123) [inline contracts/VeJoeStaking.sol:149]: tiny nit, there should be a space between `//` and `We`

File: contracts/VeJoeStaking.sol [hunk 2]
```diff
@@ -44,71 +37,43 @@ contract VeJoeStaking is Initializable, OwnableUpgradeable {
...
-    /// The only exception is the user will also receive boosted benefits if it's their first
-    /// time staking.
-    uint256 public boostedThreshold;
+    /// @notice The last time that the reward variables were updated
+    uint256 public lastRewardTimestamp;
 
-    /// @notice The length of time a user receives boosted benefits
-    uint256 public boostedDuration;
+    /// @notice Rate of veJOE generated per sec per JOE staked, scaled to `GENERATION_RATE_PRECISION`
+    uint256 public baseGenerationRate;
```
Comment (cryptofish7) [inline contracts/VeJoeStaking.sol:50]: If we're removing `boostedGenerationRate`, is there any point to having `baseGenerationRate`?

File: contracts/VeJoeStaking.sol [hunk 3]
```diff
@@ -245,13 +147,14 @@ contract VeJoeStaking is Initializable, OwnableUpgradeable {
             "VeJoeStaking: cannot withdraw greater amount of JOE than currently staked"
         );
 
+        updateRewardVars();
+        // Note that we don't need to claim as the user's veJOE balance will be reset to 0
+
         userInfo.balance = userInfo.balance.sub(_amount);
```
Comment (cryptofish7) [inline contracts/VeJoeStaking.sol:153]:
```
        updateRewardVars();

        // Note that we don't need to claim as the user's veJOE balance will be reset to 0
        userInfo.balance = userInfo.balance.sub(_amount);
```

File: contracts/VeJoeStaking.sol [hunk 4]
```diff
@@ -262,98 +165,65 @@ contract VeJoeStaking is Initializable, OwnableUpgradeable {
...
+        updateRewardVars();
         _claim();
     }
 
     /// @notice Get the pending amount of veJOE for a given user
     /// @param _user The user to lookup
-    /// @return The number of pending veJOE tokens for `_user`
-    function getPendingVeJoe(address _user) public view returns (uint256) {
+    /// @return pendingVeJoe The number of pending veJOE tokens for `_user`
+    function getPendingVeJoe(address _user) public view returns (uint256 pendingVeJoe) {
```
Comment (cryptofish7) [inline contracts/VeJoeStaking.sol:175]: Any reason why you choose to declare the return variable here?

**Review by cryptofish7 — CHANGES_REQUESTED** [review]

File: contracts/VeJoeStaking.sol (hunk 4 above)
Comment (LouisMeMyself) [reply contracts/VeJoeStaking.sol:175]: Indeed better not to, as we don't always return `pendingVeJoe`

File: contracts/VeJoeStaking.sol (hunk 2 above)
Comment (cryptofish7) [reply contracts/VeJoeStaking.sol:50]: Ah yes good point. In that case, let's rename to `veJoePerSec`

**Review by cryptofish7 — CHANGES_REQUESTED** [review]

**Review by cryptofish7 — APPROVED** [review]

**Pull request merged**